packages/snaps-utils/src/snaps.test.ts

```diff
@@ -192,3 +197,56 @@ describe('HttpSnapIdStruct', () => {
     expect(is(value, HttpSnapIdStruct)).toBe(false);
   });
 });
+
+describe('hasSnap', () => {
+  it("will check an origin's permissions object to see if it has permission to interact with a specific snap", () => {
+    const permissionKey = 'wallet_snap';
+    const validPermissions = {
+      [permissionKey]: {
+        date: 1,
+        id: '1',
+        invoker: 'example.com',
+        parentCapability: 'wallet_snap',
+        caveats: [
+          {
+            type: 'snapIds',
+            value: {
+              foo: {},
+            },
+          },
+        ],
+      },
+    } as SubjectPermissions<PermissionConstraint>;
```

Comment:
Would this work in this case?
```suggestion
    const validPermissions: SubjectPermissions<PermissionConstraint> = {
      [permissionKey]: {
        date: 1,
        id: '1',
        invoker: 'example.com',
        parentCapability: 'wallet_snap',
        caveats: [
          {
            type: 'snapIds',
            value: {
              foo: {},
            },
          },
        ],
      },
    };
```